block_util/src/raw_sync.rs

```diff
@@ -65,15 +60,19 @@ impl AsyncIo for RawFileSync {
         iovecs: Vec<libc::iovec>,
         user_data: u64,
     ) -> AsyncIoResult<()> {
-        read_vectored_sync(
-            offset,
-            iovecs,
-            user_data,
-            &mut self.raw_file,
-            &self.eventfd,
-            &mut self.completion_list,
-            &mut self.semaphore,
-        )
+        let result = unsafe {
+            libc::preadv(
+                self.fd as libc::c_int,
+                iovecs.as_ptr() as *const libc::iovec,
+                iovecs.len() as libc::c_int,
+                offset,
+            )
+        };
+
```

Comment:
Are you ignoring the errors here? i.e. result = -1 and errno set.

Reply:
No I'm pushing the result down the completion list, so that the caller will get the result after calling into `complete()`

Reply:
But errno might have been overwritten. And this is not what the old code did:

https://github.com/cloud-hypervisor/cloud-hypervisor/blob/060df69f072986ef5a7f15a4bde59c3da75b5403/block_util/src/lib.rs#L583